pkg/render/intrusion_detection.go

```diff
@@ -892,3 +950,243 @@ func (c *intrusionDetectionComponent) intrusionDetectionAnnotations() map[string
 	}
 	return nil
 }
+
+// intrusionDetectionComponentResources sets the default requirements value for DPI in IDS resource if it is not set.
+func (c *intrusionDetectionComponent) intrusionDetectionComponentResources() {
+	if c.ids.Spec.ComponentResources == nil {
+		c.ids.Spec.ComponentResources = []operator.IntrusionDetectionComponentResource{
+			{
+				ComponentName: operator.ComponentNameDeepPacketInspection,
+				ResourceRequirements: &corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						corev1.ResourceMemory: resource.MustParse(DefaultMemoryLimitDPI),
+						corev1.ResourceCPU:    resource.MustParse(DefaultCPULimitDPI),
+					},
+					Requests: corev1.ResourceList{
+						corev1.ResourceMemory: resource.MustParse(DefaultMemoryRequestDPI),
+						corev1.ResourceCPU:    resource.MustParse(DefaultCPURequestDPI),
+					},
+				},
+			},
+		}
+	}
+}
+
+func (c *intrusionDetectionComponent) dpiDaemonset() *appsv1.DaemonSet {
+	var terminationGracePeriod int64 = 0
+
+	return &appsv1.DaemonSet{
+		TypeMeta: metav1.TypeMeta{Kind: "DaemonSet", APIVersion: "apps/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      DeepPacketInspectionName,
+			Namespace: DeepPacketInspectionNamespace,
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"k8s-app": DeepPacketInspectionName}},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"k8s-app": DeepPacketInspectionName,
+					},
+				},
+				Spec: v1.PodSpec{
+					Tolerations:                   rmeta.TolerateAll,
+					ImagePullSecrets:              secret.GetReferenceList(c.pullSecrets),
+					ServiceAccountName:            DeepPacketInspectionName,
+					TerminationGracePeriodSeconds: &terminationGracePeriod,
+					HostNetwork:                   true,
+					Containers:                    []v1.Container{c.dpiContainer()},
+					Volumes:                       c.dpiVolumes(),
+				},
+			},
+		},
+	}
+}
+
+func (c *intrusionDetectionComponent) dpiContainer() v1.Container {
+	privileged := false
+	// On OpenShift Snort needs privileged access to access host network
+	if c.openshift {
+		privileged = true
+	}
+
+	return v1.Container{
+		Name:         DeepPacketInspectionName,
+		Image:        c.dpiImage,
+		Resources:    *c.ids.Spec.ComponentResources[0].ResourceRequirements,
+		Env:          c.dpiEnvVars(),
+		VolumeMounts: c.dpiVolumeMounts(),
+		SecurityContext: &corev1.SecurityContext{
+			Privileged: &privileged,
+		},
+		LivenessProbe: c.dpiReadinessProbes(),
```

Comment:
What about the readiness probe ?

Reply:
I've updated this to be a readiness probe (I'll make corresponding changes to DPI repository), there is no liveness probe, dpi pod panics if it encounters an irrecoverable issue so it does not necessarily need a liveness probe.